crates/miden-objects/src/note/note_tag.rs

```diff
@@ -248,23 +270,55 @@ impl fmt::Display for NoteTag {
 
 impl From<u32> for NoteTag {
     fn from(value: u32) -> Self {
-        Self(value)
+        let prefix = value >> 30;
+        let lower_30_bits = value & 0x3FFFFFFF; // Mask to get the lower 30 bits
+
+        match prefix {
+            0b00 => {
+                // Network, Specific Account (lower 30 bits are the account ID)
+                Self::NetworkAccount(lower_30_bits)
+            },
+            0b01 => {
+                // NetworkUse case (split lower 30 bits into two u16)
+                let use_case = (lower_30_bits >> 16) as u16;
+                let payload = lower_30_bits as u16;
+                Self::NetworkUseCase(use_case, payload)
+            },
+            0b10 => {
+                // Local, Any (lower 30 bits are the account ID)
+                Self::LocalAccount(lower_30_bits)
+            },
+            0b11 => {
```

Comment:
I would use the constants here instead of hardcoding the bits values.

And depending on https://github.com/0xMiden/miden-base/issues/1321#issuecomment-2850792030 I would also not mask out anything but include the value directly (or split it into two `u16`s for the use case variants).